### DISCUSSION TITLE
Clients for other languages

### DISCUSSION THREAD

**Discussion by sejr** [sejr/firegraph#37] [Ideas]
Node / TypeScript was an obvious first choice for Firegraph, but Firestore obviously supports many other languages.

My priorities would be:

1. Golang
2. Swift
3. Java
4. etc

**Comment by lcretan** [comment]
Standard Cloud Firestore Gen1 (/ Datastore) was built on top of Cloud Spanner Complex with tricks, as we can read in the paper of Google.

Enterprise Cloud Firestore is partially compatible with MongoDB; however, Gen2 is still under construction, and Gen2 has lost some of Gen1's great points.

I think that the full-featured GraphQL implementation is one of the trials for removing Gen1's small limitations as a higher-grade NoSQL DB, also you added "easy-to-use, small footprint"

Could you port Cloud Firestore Gen1 SDK and Firegraph for Perl 5?
